### PR TITLE
Bring `kanagawa` colours better in line with neovim version

### PR DESCRIPTION
**Pull request by kareigu**
- Fixes some colours not matching their counterpart in neovim.
- Adds `ui.debug` colours
- Fix for separators in inactive statuslines

### CODE REVIEW

**Comment by the-mikedavis** [comment]
\cc @zetashift what do you think?

**Comment by zetashift** [comment]
Everything looks pretty good to me, and indeed it does look more faithful to the neovim version:
image

Though personally I'm not a fan of the italics for things like `defmodule` and `def`, but I can change that myself!

**Comment by the-mikedavis** [comment]
Hmm yeah I agree the italics look a bit odd on keywords. It does look accurate to the upstream though, at least the screenshots in https://github.com/rebelot/kanagawa.nvim

File: runtime/themes/kanagawa.toml [hunk 1]
```diff
@@ -31,11 +30,11 @@
 "ui.bufferline.active" = { fg = "oldWhite", bg = "sumiInk0" }
 "ui.bufferline.background" = { bg = "sumiInk0" }
 
-"ui.popup" = { fg = "fujiWhite", bg = "sumiInk0" }
+"ui.popup" = { fg = "fujiWhite", bg = "sumInk0" }
```
Comment (the-mikedavis) [inline runtime/themes/kanagawa.toml:33]:
```suggestion
"ui.popup" = { fg = "fujiWhite", bg = "sumiInk0" }
```

looks like a typo here

**Comment by kareigu** [comment]
I prefer them without as well but as upstream sets italics as well I thought it better to just follow that by default.

**Comment by luetage** [comment]
@kareigu I haven’t built Helix for a while, therefore I didn’t do any more pull requests for Kanagawa. What’s missing is colors for `ui.debug.breakpoint` and `ui.debug.active`. Additionally completing markdown. Please consider adding them.

```
"ui.debug.breakpoint" = "peachRed"
"markup.list" = "oniViolet"
```

**Comment by luetage** [comment]
Checked it out, all looks fine, great job. What I don’t like is the added gutter background color, but I don’t like that on any Helix theme it is used on. And by changing sumiInk4 the newline character (ui.virtual) becomes pretty much invisible (but that’s also a matter of taste of course).

One thing we could include is a fix for statusline separators: `"ui.statusline.separator" = { fg = "", bg = "" }`
There’s a way to theme them but only for the active statusline, this workaround fixes the problem with separators in inactive statuslines. Something to think about.

**Comment by kareigu** [comment]
> And by changing sumiInk4 the newline character (ui.virtual) becomes pretty much invisible (but that’s also a matter of taste of course).

Ah, I was hoping nothing broke because of that since I wasn't seeing anything :D
I'll look into that. The real issue here was that sumiInk4 value was previously set to the sumiInk6 value.

**Comment by kareigu** [comment]
> One thing we could include is a fix for statusline separators: `"ui.statusline.separator" = { fg = "", bg = "" }`
> There’s a way to theme them but only for the active statusline, this workaround fixes the problem with separators in inactive statuslines. Something to think about.

Tested this out and it's way better, thanks.

**Review by zetashift — APPROVED** [review]
> I prefer them without as well but as upstream sets italics as well I thought it better to just follow that by default.

Kanagawa does provide some code to easily disable italics for certain treesitter nodes. But I agree here, go with the default.

> I can't be the only one who sees the URLs with treesitter as utterly unreadable, so I've gone with the default highlighting version.

I'm okay with the default vim highlighting colors.

**Review by the-mikedavis — APPROVED** [review]

**Pull request merged**